### PR TITLE
Pass through method return type when using a method call to fetch mock

### PR DESCRIPTION
**Pull request by timheilig**
ex. Phake::when($this->getMock())->methodName();

Also fixes issue where fully qualified classes would not be found in the
index.

### CODE REVIEW

**Comment by huggsboson** [comment]
I have to say that this is extremely important in increasing adoption when using mockito and other like minded mocking libraries.

File: src/com/digitalsandwich/phake/PhakeMockTypeProvider.java [hunk 1]
```diff
@@ -146,6 +137,39 @@ else if (s.substring(0, 4).equals(CALLTYPE_STUBBED_METHOD))
         return signedClasses.size() == 0 ? null : signedClasses;
     }
 
+    private Collection<? extends PhpNamedElement> getSignedClassesFromCalltypeMock(String s, PhpIndex phpIndex) {
+        Collection<PhpNamedElement> signedClasses = new ArrayList<>();
```
Comment (huggsboson) [inline src/com/digitalsandwich/phake/PhakeMockTypeProvider.java]: 1.7 feature.  Use ArrayList&lt;PhpNamedElement&gt;

File: src/com/digitalsandwich/phake/PhakeMockTypeProvider.java [hunk 2]
```diff
@@ -109,32 +109,23 @@ private String passThruMethodParameterType(MethodReference methodReference, int
 
     @Override
     public Collection<? extends PhpNamedElement> getBySignature(String s, Project project) {
-
         PhpIndex phpIndex = PhpIndex.getInstance(project);
-        Collection<PhpNamedElement> signedClasses = new ArrayList<PhpNamedElement>();
+        Collection<PhpNamedElement> signedClasses = new ArrayList<>();
```
Comment (huggsboson) [inline src/com/digitalsandwich/phake/PhakeMockTypeProvider.java]: 1.7 feature.  Use ArrayList&lt;PhpNamedElement&gt;

File: src/com/digitalsandwich/phake/PhakeMockTypeProvider.java [hunk 3]
```diff
@@ -146,6 +137,39 @@ else if (s.substring(0, 4).equals(CALLTYPE_STUBBED_METHOD))
...
+        signedClasses.addAll(phpIndex.getBySignature(phakeSignature));
+
+        String pipeDelimitedPhpClassNames = s.substring(separator + 1);
+        signedClasses.addAll(getSignedClassesFromPipeDelimitedClassNames(pipeDelimitedPhpClassNames, phpIndex));
+        return signedClasses;
+    }
+
+    private Collection<? extends PhpNamedElement> getSignedClassesFromPipeDelimitedClassNames(
+            String pipeDelimitedPhpClassNames, PhpIndex phpIndex) {
+        Collection<PhpNamedElement> signedClasses = new ArrayList<>();
```
Comment (huggsboson) [inline src/com/digitalsandwich/phake/PhakeMockTypeProvider.java]: 1.7 feature.  Use ArrayList&lt;PhpNamedElement&gt;

**Comment by timheilig** [comment]
Digging around the openapi I figured out how to make a much smaller change, pushing it up shortly

**Pull request closed without merging**